src/hooks/useMutation.ts

```diff
@@ -0,0 +1,66 @@
+import { useCallback, useState } from "react";
+
+interface Options {
+  variables: Record<string, string | number | boolean>;
+}
+
+interface ErrorBody {
+  status: number;
+  statusText: string;
+  message: string;
+}
+
+export function useMutation<TResponse>(
+  url: string,
+  method: "POST" | "PUT" | "PATCH" | "DELETE",
```

Comment:
I think this hook is pretty much duplicating` useFetch()`. I think it's better to merge them both. Could you remove `useEffect` from the hook(`useFetch`) and return `makeRequest()` like this hook does?
This way you can remove this custom hook and `useFetch()` will be more flexible, which will achieve the same as this hook tires to do.

Reply:
@kei95 
I thought about that way, in that case we have to write fetch like below.

```
const {makeRequest, ...} = useFetch(FETCH_URL(id));
const project = await makeRequest();
```
Do you have any idea that we can write one line for fetch? Yes, useMutation is from GraphQL term... lol

Reply:
As a record, `useFetch()` and this hooks ain't too duplicating each other. Thus we'll keep these hooks for when we want to make requests on demand👍